### PR TITLE
225 add an about page

### PR DESCRIPTION
**Pull request by shrprabh**
This pull request includes several changes to the `src/components`, `src/layout`, and `src/routes` directories to add a new `About` page and improve type safety in the `CopyToClipboardButton` component. The most important changes include adding the `About` page component, updating the main layout to include a link to the `About` page, and adding type definitions to the `CopyToClipboardButton` component.

### New `About` Page:

* [`src/components/About.tsx`](diffhunk://#diff-8153b95ba37a69296bb9a433eabaafdbcec247f70529d101574ef340b2f032fbR1-R201): Added a new `About` page component that includes detailed information about BilboMD, its features, and pipeline options.

### Main Layout Updates:

* [`src/layout/MainLayout/index.tsx`](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcR129-R135): Added an `About` menu item to the main layout navigation, allowing users to access the new `About` page.
* [`src/layout/MainLayout/index.tsx`](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcL17-R18): Corrected the import statement for `InfoOutlined` icon.

### Route Updates:

* [`src/routes/MainRoutes.tsx`](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R89-R92): Added a new route for the `About` page under the `dashboard` path.
* [`src/routes/MainRoutes.tsx`](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R18-R19): Imported the `About` component to be used in the new route.

### Type Safety Improvements:

* [`src/components/Common/CopyToClipboardButton.tsx`](diffhunk://#diff-8b590fe07d9901337e5185b86ecf10cc5993670fa368c9876d4740189114d58fL5-R9): Added TypeScript interface for `CopyToClipboardButton` component props to improve type safety.
image

### CODE REVIEW

**Comment by shrprabh** [comment]
@dsclassen  In an attempt to repurpose the home component and direct it as an about component, I discovered that the home component was redirecting to the welcome component after token verification. To enhance clarity, I introduced a new component. If necessary, we can relocate this component to a different directory.

**Comment by shrprabh** [comment]
there are few changes i need to add to include current features will be pushing the changes.

**Comment by dsclassen** [comment]
I do remember this being an issue... unauthenticated users get the `src/components/Home.tsx` component. Authenticated users get bounced to `src/features/auth/Welcome.tsx`.

I think it makes good sense to have a dedicated `About.tsx` component and it can certainly reuse stuff from `Home`

**Comment by dsclassen** [comment]
I would like to not add new code to the `components` folder.  In fact I would like to slowly move everything out of that directory into `features`. Perhaps a new directory `features/about` ?

**Comment by shrprabh** [comment]
Thank you for your feedback. I will relocate the “about” to **features directory**. Additionally, I have created four reusable components by creating a shared directory under components directory to enhance reusability and facilitate unit testing. I will be moving that as well to the “features” directory. These components can be utilized in the “home” component as well. **If this is acceptable,** I can incorporate them into the “home” component immediately. I have not utilized the shared components in the “home” component, but I have used them in the “about” component.


### New Components [Shared Components ]:
image

* [`src/components/About.tsx`](diffhunk://#diff-8153b95ba37a69296bb9a433eabaafdbcec247f70529d101574ef340b2f032fbR1-R221): Introduced a new `About` component that provides detailed information about BilboMD, including features, pipeline options, and additional resources.
* [`src/components/Shared/AdditionalInfo.tsx`](diffhunk://#diff-d2dc169846e8311e8d0f0eff357961eaa780bf94d6b51457928ae130a4ed8cf2R1-R15): Added a new `AdditionalInfo` component to display information about ongoing development and provide a link to the GitHub repository for feedback.
* [`src/components/Shared/FeaturesList.tsx`](diffhunk://#diff-0524be68894438ef4c45cbbcf163845c4fd1cf87f3616409ad954b7bdaa5ec10R1-R16): Created a `FeaturesList` component to list the current features of BilboMD.
* [`src/components/Shared/Introduction.tsx`](diffhunk://#diff-0341b67e787baf5c052975d82a7d995f0de0bb466dbc413ab788f155cb33c8caR1-R19): Added an `Introduction` component to provide an introductory section for the `About` page.
* [`src/components/Shared/PipelineOptions.tsx`](diffhunk://#diff-25022d047ec93576f33068044d5079741cf224a816fed830c99de329cb2182b1R1-R69): Introduced a `PipelineOptions` component to display different pipeline options available in BilboMD.

### TypeScript Enhancements:

* [`src/components/Common/CopyToClipboardButton.tsx`](diffhunk://#diff-8b590fe07d9901337e5185b86ecf10cc5993670fa368c9876d4740189114d58fL5-R9): Updated the `CopyToClipboardButton` component to use TypeScript by defining `CopyToClipboardButtonProps` interface.

### Routing and Layout Updates:

* [`src/layout/MainLayout/index.tsx`](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcL17-R18): Updated the layout to include a new "About" menu item that navigates to the `About` page. [[1]](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcL17-R18) [[2]](diffhunk://#diff-1c9d05574c39ab6daddab5c3e068d794eff4e43e682bb1cb2c9e911cad9eb2fcR129-R135)
* [`src/routes/MainRoutes.tsx`](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R18-R19): Added routing for the new `About` page under the dashboard route. [[1]](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R18-R19) [[2]](diffhunk://#diff-e6c56904d47ef4a311c46b645f2ed706df133b33912283034072a08ef8681e93R89-R92)
image

**Comment by dsclassen** [comment]
yes, please complete your refactor by incorporating the new components into the `Home` component.

**Comment by shrprabh** [comment]
Hello @dsclassen, I’ve refactored the code and moved the about component to the features directory. I’ve verified the code in light and dark modes, including login verification, loaders, and authentication. Please review the code and let me know if it’s simple and good. If not, suggest areas for improvement. Currently, the home component is in the components directory, while about and shared components are in features. I can move home to features if needed.

File: src/features/about/About.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    ]
+
+    // Define the pipeline options
+    const pipelines = [
+      {
+        title: 'BilboMD Classic w/PDB',
+        description: 'Bring your own starting PDB model and constraints.',
+        imagePath: {
+          light: '/images/bilbomd-classic-pdb-schematic.png',
+          dark: '/images/bilbomd-classic-pdb-schematic.png'
```
Comment (dsclassen) [inline src/features/about/About.tsx:97]: your light and dark image paths are identical

File: src/features/about/About.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+import { useGetConfigsQuery } from 'slices/configsApiSlice'
+import useTitle from 'hooks/useTitle'
+import useTheme from '@mui/material/styles/useTheme'
+import Introduction from '../shared/Introduction'
+import FeaturesList from '../shared/FeaturesList'
+import PipelineOptions from '../shared/PipelineOptions'
+import AdditionalInfo from '../shared/AdditionalInfo'
+
+const About = ({ title = 'About BilboMD' }) => {
+  useTitle(title)
```
Comment (dsclassen) [inline src/features/about/About.tsx:18]: I don't think that `title` needs to be a prop

```javascript
const About = () => {
useTitle("BilboMD: About")
```

should suffice

File: src/features/about/About.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+          </li>
+          <li>
+            Automagic <b>PAE Jiffy™</b> allows you to upload AlphaFold PDB and
+            PAE JSON files from AlphaFold.
+          </li>
+        </ul>
+      </>,
+      'You can see the status and history of your BilboMD jobs.',
+      'Provides a download link to retrieve the <code>results.tar.gz</code> file.',
+      'Currently we store BilboMD job results for 60 days, so be sure to download your results if they are important to you.'
```
Comment (dsclassen) [inline src/features/about/About.tsx:87]: add <b></b> tags to BilboMD please

File: src/features/shared/AdditionalInfo.tsx
Comment (dsclassen) [inline src/features/shared/AdditionalInfo.tsx:1]: I don't think we will ever reuse this component with different values for `githubLink`
Maybe just hardcode the repo URL

File: src/routes/MainRoutes.tsx [hunk 4]
```diff
@@ -84,6 +86,10 @@ const ProtectedMainRoutes = {
                   path: 'dashboard',
                   children: [
                     { path: '', element: <Welcome /> },
+                    {
+                      path: 'about',
+                      element: <About title='About BilboMD' />
+                    },
                     {
```
Comment (dsclassen) [inline src/routes/MainRoutes.tsx:93]: I see now why you were passing `title` into `About`.
I kind of like this pattern. I may implement this throughout the entire application.
Please keep it for now.

File: src/features/about/About.tsx (hunk 2 above)
Comment (dsclassen) [reply src/features/about/About.tsx:18]: I see what you're doing here. Please disregard my comment and leave as-is. Maybe change default value to:

```
BilboMD: About
```

**Review by dsclassen — COMMENTED** [review]
please see comments

File: src/features/about/About.tsx (hunk 2 above)
Comment (shrprabh) [reply src/features/about/About.tsx:18]: Sure, Done

File: src/features/about/About.tsx (hunk 1 above)
Comment (shrprabh) [reply src/features/about/About.tsx]: Updated with proper images for all the four pipelines.

File: src/routes/MainRoutes.tsx (hunk 4 above)
Comment (shrprabh) [reply src/routes/MainRoutes.tsx:93]: Corrected

File: src/features/shared/AdditionalInfo.tsx
Comment (shrprabh) [reply src/features/shared/AdditionalInfo.tsx:1]: currently we are using it in Home.tsx and About.tsx. is it fine to hardcode it in both or shall i leave as it is         <AdditionalInfo githubLink='https://github.com/bl1231/bilbomd-ui' />

File: src/components/Home.tsx
[the remark this answers is not in the document]
Comment (shrprabh) [reply src/components/Home.tsx:1]: I have added added all the corrections.

File: src/features/about/About.tsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+            CHARMM-compatible constraint file.
+          </li>
+          <li>
+            Automagic <b>PAE Jiffy™</b> allows you to upload AlphaFold PDB and
+            PAE JSON files from AlphaFold.
+          </li>
+        </ul>
+      </>,
+      'You can see the status and history of your BilboMD jobs.',
+      'Provides a download link to retrieve the <code>results.tar.gz</code> file.',
```
[the remark this answers is not in the document]
Comment (shrprabh) [reply src/features/about/About.tsx]: corrected by using the react fragments.

File: src/features/about/About.tsx (hunk 3 above)
Comment (shrprabh) [reply src/features/about/About.tsx]: Done

File: src/features/shared/AdditionalInfo.tsx
Comment (dsclassen) [reply src/features/shared/AdditionalInfo.tsx:1]: What is better.. hardcoding in one place or in 2 places?

Comment (shrprabh) [reply src/features/shared/AdditionalInfo.tsx:1]: @dsclassen Perfect! I initially misunderstood and thought I needed to remove the additionalInfo.tsx file. However, we only need to hardcode it in additionalInfo.tsx, and we can reuse it without passing props. That’s great!. Added the changes.

File: src/layout/MainLayout/index.tsx [hunk 6]
```diff
@@ -126,6 +126,13 @@ export default function ClippedDrawer() {
       path: '/admin',
       onclick: () => navigate('admin'),
       roles: ['admin']
+    },
+    {
+      text: 'About',
+      icon: <InfoOutlined />,
+      path: '/dashboard/about',
+      onclick: () => navigate('/dashboard/about'),
+      roles: ['user', 'manager', 'admin']
```
Comment (dsclassen) [inline src/layout/MainLayout/index.tsx:135]: Our filtering logic is a bit wonky here. I can't see the `About` menu item as a user with role = User

Please change to:

```javascript
    {
      text: 'About',
      icon: <InfoOutlined />,
      path: '/dashboard/about',
      onclick: () => navigate('/dashboard/about'),
      roles: ['user']
    }
```

and while you're at it you can change all the ones that currently say:

```javascript
roles: ['user', 'manager']
```

to:

```javascript
roles: ['user']
```

**Review by dsclassen — CHANGES_REQUESTED** [review]
Just change the menu item filtering stuff and I think we're good to merge.

File: src/layout/MainLayout/index.tsx (hunk 6 above)
Comment (shrprabh) [reply src/layout/MainLayout/index.tsx]: It implies that the manager and admins are also default users.

**Review by dsclassen — APPROVED** [review]

**Pull request merged**